### ISSUE TITLE
`Array_gpu::operator=` does not allocate correctly

### ISSUE THREAD

**Issue by stijnh** [microhh/rte-rrtmgp-cpp#31]
On the current main branch, I feel that `operator=` of `Array_gpu` is not implemented correctly. It first sets `this->ncells` and then checks if `this->ncells == 0` to allocate memory if required. However, this well never happens since `ncells`  was just set beforehand:

https://github.com/microhh/rte-rrtmgp-cpp/blob/129363e14df40e00cb0e2414489ac5f0501ab915/include/Array.h#L377

I will fix this in a PR once I understand the Array class better.

**Comment by Chiil** [comment]
You are correct. I implemented the view option later and something seem to have gone wrong here. As said before. On the long term we are also open for other array classes if you have better alternatives.
